samples/dnn/edge_detection.cpp

```diff
@@ -0,0 +1,207 @@
+        cv::dnn::DNN_BACKEND_VKCOM, cv::dnn::DNN_BACKEND_CUDA, cv::dnn::DNN_BACKEND_WEBNN);
+
+    const string target_keys = format(
+        "{ target          | 0 | Choose one of target computation devices: "
+                              "%d: CPU target (by default), "
+                              "%d: OpenCL, "
+                              "%d: OpenCL fp16 (half-float precision), "
+                              "%d: VPU, "
+                              "%d: Vulkan, "
+                              "%d: CUDA, "
+                              "%d: CUDA fp16 (half-float preprocess) }",
+        cv::dnn::DNN_TARGET_CPU, cv::dnn::DNN_TARGET_OPENCL, cv::dnn::DNN_TARGET_OPENCL_FP16,
+        cv::dnn::DNN_TARGET_MYRIAD, cv::dnn::DNN_TARGET_VULKAN, cv::dnn::DNN_TARGET_CUDA,
+        cv::dnn::DNN_TARGET_CUDA_FP16);
+
+    const string keys = param_keys + backend_keys + target_keys;
+
+    CommandLineParser parser(argc, argv, keys);
+    parser.about(about);
+
+    string inputImagePath = parser.get<string>("input");
+    string onnxModel = parser.get<string>("model");
+    string method =  parser.get<string>("method");
+    int backend = parser.get<int>("backend");
+    int target = parser.get<int>("target");
+    int imageSize =  parser.get<int>("imageSize");
+
+    Mat image = imread(samples::findFile(inputImagePath));
+    if (image.empty()) {
+        cout << "Could not read the image: " << inputImagePath << endl;
+        return 1;
+    }
+
+    if (method == "dexined") {
+        Net net = readNetFromONNX(onnxModel);
+        net.setPreferableBackend(backend);
+        net.setPreferableTarget(target);
+        Mat preprocessed = preprocess(image, imageSize);
+
+        Mat blob = blobFromImage(preprocessed);
+        net.setInput(blob);
+
+        Mat result = net.forward();
+
+        vector<Mat> outputs;
+        net.forward(outputs); // Get all output layers
+        int originalWidth = image.cols;
+        int originalHeight = image.rows;
+        pair<Mat, Mat> res = postProcess(outputs, originalHeight, originalWidth);
+        Mat fusedOutput = res.first;
+        Mat averageOutput = res.second;
+
+
+        imshow("Input", image);
+        imshow("Output", fusedOutput);
+        waitKey(0);
+    } else if (method == "canny") {
+        Mat gray;
+        cvtColor(image, gray, COLOR_BGR2GRAY);
+
+        UserData user_data;
+        user_data.gray = gray;
+
+        namedWindow("Output", WINDOW_NORMAL);
+        namedWindow("Input", WINDOW_NORMAL);
+
+         // Create trackbars
+        createTrackbar("thrs1", "Output",0, 255, cannyDetectionThresh1, &user_data);
+        createTrackbar("thrs2", "Output",0, 255, cannyDetectionThresh2, &user_data);
+
+        // Set initial positions of trackbars
+        setTrackbarPos("thrs1", "Output", 100);
+        setTrackbarPos("thrs2", "Output", 200);
+
+        imshow("Input", image);
+        waitKey(0);
```

Comment:
it would be really nice to be able to switch between Canny and deep learning-based modes with a key